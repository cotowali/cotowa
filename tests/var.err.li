var i = 0
var i = 0
var n: int = 's'
n = 's'

var (t1, t2): (int, int) = ('a', 'b')
(i, n) = ('a', 'b')

var _ = 10
echo(_)

fn f() {
  n = '10'
  (i, n) = ('a', 'b')
}
