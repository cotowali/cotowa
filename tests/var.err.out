error: var.err.li 2,5: `i` is already defined
    2 | var i = 0
      |     ^
error: var.err.li 3,5: mismatched types: `int` (expected) and `string` (found)
    3 | var n: int = 's'
      |     ^^^^^^^^^^^^
error: var.err.li 4,1: mismatched types: `int` (expected) and `string` (found)
    4 | n = 's'
      | ^^^^^^^
error: var.err.li 6,5: mismatched types: `(int, int)` (expected) and `(string, string)` (found)
    6 | var (t1, t2): (int, int) = ('a', 'b')
      |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: var.err.li 7,1: mismatched types: `(int, int)` (expected) and `(string, string)` (found)
    7 | (i, n) = ('a', 'b')
      | ^^^^^^^^^^^^^^^^^^^
error: var.err.li 10,6: variable `_` is not defined
   10 | echo(_)
      |      ^
error: var.err.li 13,3: cannot assign to variables outside of current function
   13 |   n = '10'
      |   ^^^^^^^^
error: var.err.li 14,3: cannot assign to variables outside of current function
   14 |   (i, n) = ('a', 'b')
      |   ^^^^^^^^^^^^^^^^^^^
error: var.err.li 18,7: expected 2 variables, but found 3 variables
   18 |   var (a, b, c): (int, int)
      |       ^^^^^^^^^
